packages/vscode-ide-companion/src/ide-server.ts

```diff
@@ -8,28 +8,80 @@ import * as vscode from 'vscode';
 import { McpServer } from '@modelcontextprotocol/sdk/server/mcp.js';
 import { StreamableHTTPServerTransport } from '@modelcontextprotocol/sdk/server/streamableHttp.js';
 import express, { Request, Response } from 'express';
+import { randomUUID } from 'node:crypto';
+import {
+  isInitializeRequest,
+  type JSONRPCNotification,
+} from '@modelcontextprotocol/sdk/types.js';
 
 export async function startIDEServer(_context: vscode.ExtensionContext) {
   const app = express();
   app.use(express.json());
 
-  const mcpServer = createMcpServer();
-  const transport = new StreamableHTTPServerTransport({
-    sessionIdGenerator: undefined,
-    enableJsonResponse: true,
-  });
+  const transports: { [sessionId: string]: StreamableHTTPServerTransport } = {};
 
-  mcpServer.connect(transport);
+  vscode.window.onDidChangeActiveTextEditor((editor) => {
```

Comment:
iirc this returns a disposable to help cleanup state in the IDE that we then need to return as part of the normal extension bootup in VSCode